src/test/java/com/arangodb/StreamTransactionGraphTest.java

```diff
@@ -0,0 +1,396 @@
+ *
+ * Copyright holder is ArangoDB GmbH, Cologne, Germany
+ */
+
+package com.arangodb;
+
+import com.arangodb.ArangoDB.Builder;
+import com.arangodb.entity.*;
+import com.arangodb.model.*;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Collections;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * @author Michele Rastelli
+ */
+@RunWith(Parameterized.class)
+public class StreamTransactionGraphTest extends BaseTest {
+
+    private static final String GRAPH_NAME = "graph_stream_transaction_graph_test";
+    private static final String EDGE_COLLECTION = "edge_collection_stream_transaction_graph_test";
+    private static final String VERTEX_COLLECTION_1 = "vertex_collection_1_stream_transaction_graph_test";
+    private static final String VERTEX_COLLECTION_2 = "vertex_collection_2_stream_transaction_graph_test";
+
+    private ArangoGraph graph;
+    private ArangoVertexCollection vertexCollection1;
+    private ArangoVertexCollection vertexCollection2;
+    private ArangoEdgeCollection edgeCollection;
+
+    public StreamTransactionGraphTest(final Builder builder) {
+        super(builder);
+
+        graph = db.graph(GRAPH_NAME);
+
+        if (graph.exists())
+            graph.drop();
+
+        graph.create(Collections.singletonList(new EdgeDefinition().collection(EDGE_COLLECTION).from(VERTEX_COLLECTION_1).to(VERTEX_COLLECTION_2)));
+
+        vertexCollection1 = graph.vertexCollection(VERTEX_COLLECTION_1);
+        vertexCollection2 = graph.vertexCollection(VERTEX_COLLECTION_2);
+        edgeCollection = graph.edgeCollection(EDGE_COLLECTION);
+    }
+
+    @After
+    public void teardown() {
+        if (graph.exists())
+            graph.drop();
+        if (db.collection(EDGE_COLLECTION).exists())
+            db.collection(EDGE_COLLECTION).drop();
+        if (db.collection(VERTEX_COLLECTION_1).exists())
+            db.collection(VERTEX_COLLECTION_1).drop();
+        if (db.collection(VERTEX_COLLECTION_2).exists())
+            db.collection(VERTEX_COLLECTION_2).drop();
+    }
+
+    private BaseEdgeDocument createEdgeValue(String streamTransactionId) {
+        final VertexEntity v1 = vertexCollection1.insertVertex(new BaseDocument(), new VertexCreateOptions().streamTransactionId(streamTransactionId));
+        final VertexEntity v2 = vertexCollection2.insertVertex(new BaseDocument(), new VertexCreateOptions().streamTransactionId(streamTransactionId));
+        final BaseEdgeDocument value = new BaseEdgeDocument();
+        value.setFrom(v1.getId());
+        value.setTo(v2.getId());
+        return value;
+    }
+
+    @Test
+    public void getVertex() {
+        assumeTrue(isSingleServer());
+        assumeTrue(isAtLeastVersion(3, 5));
```

Comment:
Oh, one thing:
`        assumeTrue(isAtLeastVersion(3, 5));`

^-- does this needs to be instead `isAtLeastVersion(3,5,1)` ?

Reply:
We are still discussing about which db versions to support and to test against... For the moment we only support and test against the latest patch of every minor release. Eg. atm 3.3.23, 3.4.8 and 3.5.0.